### PR TITLE
minor improvement in cramers-v runtime

### LINKED ISSUE #1914: [FEAT] Runtime improvement for drift checks
**Is your feature request related to a problem? Please describe.**
We would like to have a run time of <1ms per column for 10K samples for both categorical and numerical drift. Barring that, let's try to reduce the runtime. Currently Carmers-v takes double the time of earth movers.

Profiling results at https://drive.google.com/file/d/1OtrUg48rV9jRNL9tMHCgat7KDYGVTus3/view?usp=sharing

**Describe the solution you'd like**
Some suggestions:
1. Are there arguments for these scipy functions that can be played with?
2. Are there alternative OSS implementations?
3. numba
4. intel libraries

**Describe alternatives you've considered**
Revert to PSI as default categorical drift.

**Once Done**
Update the benchmark spreadsheet

### PR DESCRIPTION
**Pull request by noamzbr**
#### Reference Issues/PRs

Resolves #1914 

(Doesn't really solve the problem, more like a 10-20% improvement)

### CODE REVIEW

File: deepchecks/utils/distribution/preprocessing.py [hunk 1]
```diff
@@ -159,8 +159,10 @@ def preprocess_2_cat_cols_to_same_bins(dist1: Union[np.ndarray, pd.Series], dist
         list of all categories that the percentages represent.
 
     """
-    categories_list = list(set(dist1).union(set(dist2)))
     dist1_counter, dist2_counter = Counter(dist1), Counter(dist2)
+    len1 = len(dist1)
+    len2 = len(dist2)
```
Comment (Nadav-Barak) [inline deepchecks/utils/distribution/preprocessing.py:164]: len2 is only used once later on

Comment (Nadav-Barak) [reply deepchecks/utils/distribution/preprocessing.py:164]: Scratch that, but there is a bug in line 192
ratio should be multiplied by size of dists

Comment (Nadav-Barak) [inline deepchecks/utils/distribution/preprocessing.py:164]:
```suggestion
    size_dist1, size_dist2 = len(dist1), len(dist2)
```

**Review by Nadav-Barak — CHANGES_REQUESTED** [review]
Require changes due to bug found

**Review by Nadav-Barak — APPROVED** [review]

**Review by matanper — APPROVED** [review]

**Pull request merged**